client/ayon_core/hosts/houdini/api/hda_utils.py

```diff
@@ -0,0 +1,414 @@
+"""Helper functions for load HDA"""
+
+import os
+import contextlib
+import uuid
+
+import ayon_api
+from ayon_api import (
+    get_project,
+    get_representation_by_id,
+    get_versions,
+    get_folder_by_path,
+    get_product_by_name,
+    get_version_by_name,
+    get_representation_by_name
+)
+from ayon_core.pipeline.load import (
+    get_representation_context,
+    get_representation_path_from_context
+)
+from ayon_core.pipeline.context_tools import get_current_project_name
+
+from ayon_core.hosts.houdini.api import lib
+
+import hou
+
+
+def is_valid_uuid(value) -> bool:
+    """Return whether value is a valid UUID"""
+    try:
+        uuid.UUID(value)
+    except ValueError:
+        return False
+    return True
+
+
+@contextlib.contextmanager
+def _unlocked_parm(parm):
+    """Unlock parm during context; will always lock after"""
+    try:
+        parm.lock(False)
+        yield
+    finally:
+        parm.lock(True)
+
+
+def get_available_versions(node):
```

Comment:
Where is this used? If it is used on multiple nodes at the same time I would recommend to do it in bulk as this may be slow when done one by one.

Reply:
It's run once when a user clicks this button in the Houdini digital asset.
![image](https://github.com/ynput/ayon-core/assets/20871534/029e4e4c-d316-4483-8189-8ca3f8db07d6)

Reply:
It's as @MustafaJafar states - it's run once, one request by the node itself on a user clicking a button. Unfortunately it's not something I can batch across multiple nodes.